machine/util/callbacks/early_stopping_callback.py

```diff
@@ -0,0 +1,94 @@
+from machine.util.callbacks import Callback
+
+
+class EarlyStoppingCallback(Callback):
+    """
+    Original callback taken from https://github.com/ncullen93/torchsample
+    Early Stopping to terminate training early under certain conditions
+    """
+
+    def __init__(self,
+                 monitor='eval_losses',
```

Comment:
I'm a little bit confused about the interaction of `monitor` and `lm_name`. What happens if you specify `eval_losses` in `monitor` but then an accuracy in `lm_name`?

Reply:
It would break haha because `current_loss` would be undefined. Do we want to add an explicit check for it ?

Reply:
Hmm, I wonder if there is a way to do this a bit less 'convoluted'.